ldapAdapterEXT-ext-impl/src/main/java/com/bosch/osmi/sw360/ldapAdapter/CustomPortalLDAPImporterImpl.java

```diff
@@ -8,20 +8,47 @@
 import com.liferay.portal.security.ldap.LDAPSettingsUtil;
 import com.liferay.portal.security.ldap.PortalLDAPImporterImpl;
 import com.liferay.portal.security.membershippolicy.OrganizationMembershipPolicyUtil;
-import com.liferay.portal.service.*;
+import com.liferay.portal.service.OrganizationLocalService;
+import com.liferay.portal.service.OrganizationLocalServiceUtil;
+import com.liferay.portal.service.UserLocalServiceUtil;
 import org.apache.log4j.Logger;
 
 import javax.naming.NamingException;
 import javax.naming.directory.Attribute;
 import javax.naming.directory.Attributes;
 import javax.naming.ldap.LdapContext;
-import java.util.Collections;
-import java.util.Properties;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import java.util.stream.Collectors;
 import java.util.stream.LongStream;
 
+import static org.apache.log4j.LogManager.getLogger;
+
 public class CustomPortalLDAPImporterImpl extends PortalLDAPImporterImpl {
+	private static final String MAPPING_KEYS_PREFIX = "mapping.";
+	private static final String MAPPING_VALUES_SUFFIX = ".target";
+	private boolean matchPrefix = false;
+	private Map<String, String> organizationMappings;
```

Comment:
Since a `Map<k,v>` is maybe implemented as `Set<Entry<k,v>>` one looses the order on the keys.

This could be a problem for prefix-matching:

```
mapA = [ "ABC" => "org1", "AB" => "org2" ]
mapB = [ "AB" => "org2', "ABC" => "org1" ]
```

yield different output for the input `"ABCD"`.

It might be an alternative to use a datastructure with ordering and order by the length of the key.